pylint/checkers/bad_chained_comparison.py

```diff
@@ -6,12 +6,12 @@
 
 from typing import TYPE_CHECKING
 
-from astroid import nodes
-
 from pylint.checkers import BaseChecker
 from pylint.interfaces import HIGH
 
 if TYPE_CHECKING:
+    from astroid import nodes
+
```

Comment:
> > However, it does impose a greater workload on developers and us maintainer. If we change something, we constantly need to make sure to double check all imports or run the risk of something failing.
> 
> I don't understand what you mean. There is the minor one-time cost of reviewing this PR (and the other one). What other workload is there?

This is the just the first change here.

I assume at the moment `nodes` is only used for type annotations so it's "safe" in the `if TYPE_CHECKING` block. What happens if you go ahead and modify / add a new check that uses `isinstance(nodes, nodes.ClassDef)` for example? You would run the tests only to notice that they fail since `nodes` isn't defined at runtime. So you have to go back, adjust the import, and run it again.

On its own maybe not that much of work but now you move that check to another file. So the imports need to change again. Keeping that in sync will be quite frustrating, even if it can be automated with dev tools.

More importantly though, it isn't really worth it IMO.

_In the example, `nodes` is probably already cached as it's used at runtime in a different checker. Guarding an import behind `if TYPE_CHECKING` for that purpose only really makes sense if you can avoid having to import a package in the first place._

Reply:
In the scenario you're proposing, you're taking a module which does not have any runtime dependency on `astroid.nodes` and adding a new runtime dependency on it. A new dependency where before there was none. Is a new module dependency a good idea? Can the feature be implemented without adding a new module dependency? These are questions that are worth taking a few seconds to think about. If the code is structured in such a way as to highlight when a new runtime module dependency is added, well that seems like a good thing to me. Currently the true cost of adding a new dependency is hidden because typechecking imports and runtime imports are not distinguished.

Pylint remains bedeviled by two severe problems: it is very slow and it is not type-correct. Fixing the type errors would help the speed problem, since it would allow compiling with Mypyc. But the typing problem is difficult due to an internal problem, namely a gnarly module structure based on circular imports.

In light of the these well-known and long-standing problems, your idea is that the code should be optimized to make it as easy as possible to add new runtime module dependencies? That is a very strange thing to prioritize. Personally I think adding a little friction to the process might be a good thing, if it can stanch the complexity.